app/models/miq_cockpit_ws_worker.rb

```diff
@@ -0,0 +1,66 @@
+require 'miq_apache'
+
+class MiqCockpitWsWorker < MiqWorker
+  require_nested :Runner
+  require_nested :Authenticator
+
+  APACHE_CONF_FILE = '/etc/httpd/conf.d/manageiq-redirects-cockpit'.freeze
+  self.required_roles = ['cockpit_ws']
+  self.maximum_workers_count = 1
+
+  def friendly_name
+    @friendly_name ||= "Cockpit Worker"
+  end
+
+  def self.can_start_cockpit_ws?
+    @supports_cockpit_ws ||= MiqCockpit::WS.can_start_cockpit_ws?
+  end
+
+  def self.should_start_worker?
+    return false unless has_required_role?
+    can_start_cockpit_ws?
+  end
+
+  def self.sync_workers
+    install_apache_proxy_config if MiqEnvironment::Command.supports_apache?
+    @workers = should_start_worker? ? 1 : 0
+    super
+  end
+
+  def self.install_apache_proxy_config
+    url = nil
+    if has_required_role? || MiqCockpitWsWorker.worker_settings[:external_url]
+      url = MiqCockpit::WS.url(MiqCockpitWsWorker.worker_settings, "localhost")
+    elsif MiqServer.my_server.has_active_role?(MiqUiWorker.required_roles)
+      url = MiqRegion.my_region.remote_cockpit_ws_url
+    end
```

Comment:
@carbonin and I are reviewing this to try to better understand it and hopefully get it merged.

Here, it looks like we are installing an apache proxy config on all appliances supporting apache with lots of logi as to what URL to use to get to the cockpit-ws.  Why can't be